apps/web/src/pages/integrations/components/UpdateIntegrationSidebarHeader.tsx

```diff
@@ -59,7 +59,7 @@ export const UpdateIntegrationSidebarHeader = ({
       return;
     }
 
-    if (provider.primary) {
+    if (shouldSetNewPrimary) {
```

Comment:
this flag consists of: 
- if the deleted integration is the primary
- and there are some other active integrations from the same channel and env

and if both are true we will show the Select Primary Integration modal